File: tests/fixtures/deadcode/swift/Fixture.swift
protocol Task {
    func run() -> Int
}

struct ReportTask: Task {
    func run() -> Int {
        return directlyUsedHelper()
    }
}

func unusedCleanupCandidate() -> Int {
    return 42
}

func directlyUsedHelper() -> Int {
    return 7
}

public struct PublicService {
    public func status() -> String {
        return "ok"
    }
}

class ViewController {
    func viewDidLoad() {
        _ = PublicService().status()
    }
}

func generatedExcludedHelper() -> Int {
    return 13
}

let dynamicMethodName = "run"

@main
struct AppMain {
    static func main() {
        let task: Task = ReportTask()
        _ = task.run()
        ViewController().viewDidLoad()
    }
}

File: tests/fixtures/deadcode/kotlin/Fixture.kt
package deadcode.fixture

import org.junit.jupiter.api.Test
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

interface Task {
    fun run()
}

class DefaultTaskFixture : org.gradle.api.DefaultTask() {
    @org.gradle.api.tasks.TaskAction
    fun execute() {}
}

private fun unusedCleanupCandidate(): Int = 42

private fun directlyUsedHelper(): Int = 7

class PublicService {
    fun status(): String = "ok"
}

@RestController
class JobRoute {
    constructor(prefix: String)

    @GetMapping("/jobs")
    fun handle(): String = "handled"

    private fun helper(): String = "internal"
}

class FixtureTests {
    @Test
    fun exercisedByTestRunner() {}
}

fun generatedExcludedHelper(): Int = 13

val dynamicMethodName = "run"

fun main() {
    val task = object : Task {
        override fun run() {
            directlyUsedHelper()
        }
    }
    task.run()
    PublicService().status()
    JobRoute().handle()
}
